File: C++/Daily Coding Problem#7.cpp
#include<bits/stdc++.h>
using namespace std;

int countWays(string s){
    int cache[s.size() + 1];
    cache[s.size()] = 1;
    for(int i = s.size()-1;i>=0;i--){
        if(s[i] == '0'){
            cache[i] = 0;
        }else if(i == s.size() -1){
            cache[i] = 1;
        }
        if(stoi(s.substr(i,2)) <=26){
            cache[i] = cache[i+1] + cache[i+2];
        }else {
            cache[i] = cache[i+1];
        }
    }
    return cache[0];
}

int main(){
    string s;
    cout<<"Enter a String";
    cin>>s;
    cout<<countWays(s);
}

File: Javascript/Daily Coding Problem#7.js
const countWays = (s) => {
    let cache = new Array(s.length + 1);
    cache[s.length] = 1;
    for (let i = s.length - 1; i >= 0; i--) {
        if (s[i] == '0') {
            cache[i] = 0;
        } else if (i == s.length - 1) {
            cache[i] = 1;
        }
        else if (parseInt(s.substr(i, 2)) <= 26) {
            cache[i] = cache[i + 1] + cache[i + 2];
        } else {
            cache[i] = cache[i + 1];
        }
    }
    return cache[0];
}

console.log(countWays("101"));